app/models/course/assessment/answer.rb

```diff
@@ -79,8 +79,14 @@ def validate_consistent_grade
   #
   # @return [Course::Assessment::Answer::AutoGrading]
   def ensure_auto_grading!
-    create_auto_grading! unless auto_grading
+    new_auto_grading = auto_grading || create_auto_grading
+    if new_auto_grading.errors[:answer_id].empty?
```

Comment:
@allenwq I can't do that, because I can't deterministically trigger the race condition... the lines would not be covered.